storage/rocksdb/CMakeLists.txt

```diff
@@ -1,27 +1,46 @@
 # TODO: Copyrights
 
-IF (NOT EXISTS "${CMAKE_SOURCE_DIR}/rocksdb/Makefile")
+IF (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/rocksdb/Makefile")
   MESSAGE(SEND_ERROR "Missing Makefile in rocksdb directory. Try \"git submodule update\".")
 ENDIF()
 
+IF (NOT WITH_ROCKSDB)
+  MESSAGE(STATUS "Not building with MyRocks storage engine, you must specify -DWITH_ROCKSDB=1 to cmake in order to build with MyRocks.")
+  RETURN()
+ELSE ()
+  MESSAGE(STATUS "Building with MyRocks storage engine")
+ENDIF ()
+
+## adds a compiler flag if the compiler supports it
+INCLUDE(CheckCCompilerFlag)
+INCLUDE(CheckCXXCompilerFlag)
+
+# pick language dialect
+check_cxx_compiler_flag(-std=c++11 HAVE_STDCXX11)
+IF (HAVE_STDCXX11)
+  SET(CMAKE_CXX_FLAGS "-std=c++11 -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS}")
+ELSE ()
+  MESSAGE(FATAL_ERROR "${CMAKE_CXX_COMPILER} doesn't support -std=c++11, you need one that does.")
+ENDIF()
+
 # get a list of rocksdb library source files
 # run with env -i to avoid passing variables
 EXECUTE_PROCESS(
-  COMMAND env -i ${CMAKE_SOURCE_DIR}/storage/rocksdb/get_rocksdb_files.sh
+  COMMAND env -i ${CMAKE_CURRENT_SOURCE_DIR}/get_rocksdb_files.sh
   OUTPUT_VARIABLE SCRIPT_OUTPUT
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 )
 # split the list into lines
 STRING(REGEX MATCHALL "[^\n]+" ROCKSDB_LIB_SOURCES ${SCRIPT_OUTPUT})
 
 INCLUDE_DIRECTORIES(
-  ${CMAKE_SOURCE_DIR}/rocksdb
-  ${CMAKE_SOURCE_DIR}/rocksdb/include
-  ${CMAKE_SOURCE_DIR}/rocksdb/third-party/gtest-1.7.0/fused-src
+  ${CMAKE_CURRENT_SOURCE_DIR}/rocksdb
+  ${CMAKE_CURRENT_SOURCE_DIR}/rocksdb/include
+  ${CMAKE_CURRENT_SOURCE_DIR}/rocksdb/third-party/gtest-1.7.0/fused-src
 )
 
 ADD_DEFINITIONS(-DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX -DOS_LINUX
-  -DZLIB)
+  -DZLIB -DPERCONA_SERVER)
```

Comment:
No, and along with the MYR issues, it is my 'goal line' to define when we have successfully addressed all import related issues. Some might be to remove the offending feature/code, some will be to pull some portion of FacebookSQL, and some will be to replace the FacebookSQL implementation with the Percona Server implementation. Once this builds and passes MTR cleanly without the -DPERCONA_SERVER and without a single `#ifdef PERCONA_SERVER //MYR???` we will be ready to move into the next stage of release availability. Ideally, I want to have most of these fixed in 5.6 providing the 5.7 counterpart implementation is similar enough to not be fixing them twice. If any issue is significantly different between 5.6 and 5.7, then it will remain unfixed in 5.6 and get addressed immediately after the 5.7 port. Without this macro and the #ifdefs, there is no way to incrementally fix build and test each item.